main.py

```diff
@@ -1 +1,147 @@
 # Secret Santa Chain Generator
+
+import csv
+import random
+import smtplib, ssl
+from email.mime.text import MIMEText
+from email.mime.multipart import MIMEMultipart
+from email.mime.image import MIMEImage
+
+# Extracts participants, emails, and exclusions from CSV file
+with open('assets/SS-participants.csv', newline='') as f:
+    reader = csv.reader(f)
+    participant_data = list(reader)
+
+# Extracts list of participants and emails from the data list
+participants = [i[1] for i in participant_data[1:]]
+names_drawn = []
+
+# Initialize seed
+SEED = 2019
+
+# Creates a string of integers using the seed that will determine the order of future random integers
+random.seed(SEED)
+s = str(random.random())
+s_index = 2
+
+# Generates a chain until it is valid
+isValid = False
+while not isValid:
+
+    # Creates duplicate list to modify
+    temp_participants = list(participants)
+    names_drawn = [i[:2] for i in participant_data[1:]]
+
+    isValid = True
+    drawn = ""
+    first_drawn = ""
+
+    for i in range(len(participants)):
+
+        # Breaks loop when the chain becomes invalid
+        if not isValid:
+            break
+
+        # Check for exclusions only after the first name has been generated
+        if len(temp_participants) < len(participants):
+            rand = drawn
+            last_index = participants.index(rand)
+
+            # The chain is invalid if all of the remaining names are in the exclusions list of the generated name
+            if all(i in participant_data[last_index + 1][1:] for i in temp_participants):
```

Comment:
I did it this way because the random name is set to the last drawn name before the loop. So the while loop will start since this name is included in the exclusions list.

Reply:
Probably not the best practice since, theres no note of that dependency and it sounds very fragile. Something that can be done is you could create a new list of `valid participants` from `temp partcipants` - `exclusion list`. 

If that new array has no length then we flag is not valid, and we can take a random index from the new `valid participants` array without having the below whole loop or `all` checks